### PR TITLE
Adding basic Type definitions

### PR DESCRIPTION
**Pull request by vitormalencar**
## Description

This PR adds initial support for Types Definitions, now we can offer some basic type checks =) 

## Motivation and Context
the idea here is to give some better support for typescript without having to touch the source code, with the type definitions we can point to the type file and give some better developer experience when using Typescript

this adds basic support to Typescript, we can do a very basic test using `tsd` in the future would be nice to move away from  `tsd` since versions above 7 will not work for this specific scenario, we can, for now, I do believe that this is the best compromise "Value/code"

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactor (non-breaking change which is refactoring/cleaning existing functionality)
- [ ] Breaking change (which would cause existing functionality to change)

### CODE REVIEW

File: src/clipboard.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { expectType } from 'tsd';
+import Clipboard from './clipboard';
+
+expectType<Clipboard>(new Clipboard('.btn'));
```
Comment (helderburato) [inline src/clipboard.test-d.ts:4]: Just remember to add EOF.

**Review by helderburato — APPROVED** [review]
LGTM

File: src/clipboard.test-d.ts (hunk 1 above)
Comment (vitormalencar) [reply src/clipboard.test-d.ts]: done

**Review by obetomuniz — APPROVED** [review]
Sounds perfect! Great job ⚡️

**Pull request merged**